Swahilipot-Hub-Developers/industrial-attachment
# Nickson simiyu 2024

Added a new `ActivityFeed` component to display recent repository activities as part of the Hacktoberfest project. This includes creating a new` /activityfeed` page, setting up GitHub API integration, and linking the Activity Feed from the homepage.


Verifies that the ActivityFeed component correctly loads data from the GitHub API.

Steps:
- Set up .env.local with a valid `NEXT_PUBLIC_GITHUB_TOKEN.`
- Start the server with npm run dev.
- Visit the /activityfeed page.
- Expected outcome: Activity feed should load with recent repository events.
- Observed outcome: Activity feed loaded correctly, showing user images and activity descriptions.

Tests link navigation from the homepage to `/activityfeed` page.

Steps:
- Start the Next.js server.
- Go to the homepage and click “Repository Activity Feed.”
- Expected outcome: Link navigates to the `/activityfeed` page without errors.
- Observed outcome: Link navigation works as expected.

![Screenshot (101)](https://github.com/user-attachments/assets/49b27184-7a86-41a9-8d24-9d1fe682f232)
![Screenshot (103)](https://github.com/user-attachments/assets/51dba038-8ccc-472c-b74b-430634d16c0a)
